src/PoolManager.sol

```diff
@@ -105,19 +104,23 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
         override
         returns (int24 tick)
     {
-        if (key.fee.isStaticFeeTooLarge()) revert FeeTooLarge();
-
         // see TickBitmap.sol for overflow conditions that can arise from tick spacing being too large
         if (key.tickSpacing > MAX_TICK_SPACING) revert TickSpacingTooLarge();
         if (key.tickSpacing < MIN_TICK_SPACING) revert TickSpacingTooSmall();
         if (key.currency0 >= key.currency1) revert CurrenciesOutOfOrderOrEqual();
         if (!key.hooks.isValidHookAddress(key.fee)) revert Hooks.HookAddressNotValid(address(key.hooks));
 
+        uint24 swapFee = 0;
+        // Initial swap fee on dynamic fee pools is set to 0, unless updateDynamicSwapFee is called in the afterInitialize hook
```

Comment:
i guess the other option is to add another parameter `initialDynamicFee`

It's a bit ugly and might be unused the majority of the time, but it would make dynamic-fee-pool-creation cheaper? 

leaning slightly towards the extra parameter because the thought of an afterInitialize call seems like a bunch of overhead for a simple operation

Reply:
oh another option is encoding the initial dynamic fee in the bottom 20 bits of `key.fee`

Reply:
>oh another option is encoding the initial dynamic fee in the bottom 20 bits of key.fee

OH YEEEEEEAH thats a great idea! I'll do that!

Reply:
i think we can have people pass in whatever fee. If the bit is on, its dynamic fee, if its off, its static fee.
And then we can _store_ `DYNAMIC_FEE_FLAG` in the slot0 to show its a dynamic fee pool?

Reply:
if the top 4 bits 'key.fee' (0x800000) can signal that its a dynamic fee pool, i think its redundant to also store it in slot0